web/src/buddy_data.ts

```diff
@@ -204,6 +205,7 @@ export function info_for(user_id: number): BuddyUserInfo {
         name: person.full_name,
         user_id,
         status_emoji_info,
+        profile_picture: people.small_avatar_url_for_person(person),
```

Comment:
This function can cost a bit to call
```
export function gravatar_url_for_email(email: string): string {                                         
    const hash = md5(email.toLowerCase());                                                              
    return "https://secure.gravatar.com/avatar/" + hash + "?d=identicon";                               
}                                                                                                       
                                                                                                        
export function small_avatar_url_for_person(person: User): string {                                     
    if (person.avatar_url) {                                                                            
        return person.avatar_url;                                                                       
    }                                                                                                   
                                                                                                        
    if (person.avatar_url === null) {                                                                   
        return gravatar_url_for_email(person.email);                                                    
    }                                                                                                   
                                                                                                        
    return `/avatar/${person.user_id}`;                                                                 
}                                                                                                       
```

Because it does an md5 hash, which are pretty fast but not if you do like 30k in a loop. It may be possible to check with profiling, but I'm guessing the fix is to just save the value once we've computed it, and make sure doing so doesn't break anything in the settings UI for your own account (it shouldn't).
``` diff
diff --git a/web/src/people.ts b/web/src/people.ts
index 851b5a774a..da144d9971 100644
--- a/web/src/people.ts
+++ b/web/src/people.ts
@@ -811,7 +811,8 @@ export function small_avatar_url_for_person(person: User): string {
     }
 
     if (person.avatar_url === null) {
-        return gravatar_url_for_email(person.email);
+        person.avatar_url = gravatar_url_for_email(person.email);
+        return person.avatar_url;
     }
 
     return `/avatar/${person.user_id}`;
```

Reply:
So that'll load avatars for users once per page load?

I'm assuming we don't care too much about another user updating their avatar and that not being updated until the page is refreshed.

This would still potentially be somewhat slow when first opening a channel with a lot of users in the buddy list, or when searching the buddy list. Are we okay with that? 

We could also asynchronously load the photos for all users on page load in the background? If they have this setting on.

Reply:
Read the code more carefully. This is just doing the `md5` hash to compute gravatar URLs once per user per page load. It does not load avatar images themselves, just computes what URL to use to request them.

Reply:
Oh I see, it's generating the "default" photo if they don't have their own personal one set. Same question stands for if we're okay with it still being slow on the first computation (loading a stream, or searching users).

Reply:
I added a prep commit to cache the gravatar url, and double checked that the gravatar source link is still accurate and that you can still change your photo.